common/src/lib.rs

```diff
@@ -12,67 +12,61 @@ pub type Timestamp = u64;
 pub type BlockHeight = u64;
 pub type FinalizationProof = Vec<(PublicKey, TypedSignature<BlockHeader>)>;
 
-/// The state that is directly recorded in the header.
+/// All about the delegation status, which will be stored in the blockchain state.
 ///
-/// This state affects the consensus, unlike the ordinary state (which is used for data recording).
+/// Note that this is not a part of `EssentialState`.
+#[allow(clippy::type_complexity)]
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
-pub struct EssentialState {
-    /// The original validator set information before the delegation calculation.
+pub struct DelegationState {
+    /// The original validator set for this block.
     ///
     /// The order here is the leader selection priority order of the validators
-    /// which directly affects the result of `calculate_net_validator_set()`.
-    pub validator_set: Vec<(PublicKey, VotingPower)>,
-    /// The protocol version that must be used from next block.
+    /// which directly affects the effective validator set.
     ///
-    /// It must be a valid semantic version (e.g., `0.2.3`).
-    pub version: String,
-    /// The delegation state (delegator, delegatee) where indices to `validator_set` is given.
-    pub delegation: Vec<(usize, usize)>,
+    /// The `(usize, usize)` is `(first delegatee, current delegatee)`.
```

Comment:
I think it's better to add a structure here.
`validator.2` is hard to understand.

Reply:
Will fix at the end of the PR update (otherwise it might confuse other reviewer's fix request)

Reply:
fixed